kyuubi-server/src/main/scala/org/apache/kyuubi/engine/YarnApplicationOperation.scala

```diff
@@ -63,7 +65,18 @@ class YarnApplicationOperation extends ApplicationOperation with Logging {
         } else {
           try {
             val applicationId = reports.get(0).getApplicationId
-            yarnClient.killApplication(applicationId)
+            appMgrInfo.proxyUser match {
+              case Some(proxyUser) =>
+                UserGroupInformation
+                  .createProxyUser(proxyUser, UserGroupInformation.getCurrentUser)
+                  .doAs(new PrivilegedExceptionAction[Unit]() {
+                    override def run(): Unit = {
+                      yarnClient.killApplication(applicationId)
```

Comment:
UGI of `yarnClient` is set when we call `yarnClient.start()`.  `yarnClient.killApplication` will always run as the UGI set at start.